akka-stream/src/main/scala/akka/stream/javadsl/Restart.scala

```diff
@@ -36,10 +37,13 @@ object RestartSource {
    * @param randomFactor after calculation of the exponential back-off an additional
    *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
    *   In order to skip this additional delay pass in `0`.
+   * @param onlyOnFailures if true only restarts in failures and not in completions. Default false.
    * @param sourceFactory A factory for producing the [[Source]] to wrap.
+   *
    */
-  def withBackoff[T](minBackoff: FiniteDuration, maxBackoff: FiniteDuration, randomFactor: Double, sourceFactory: Creator[Source[T, _]]): Source[T, NotUsed] = {
-    akka.stream.scaladsl.RestartSource.withBackoff(minBackoff, maxBackoff, randomFactor) { () ⇒
+  def withBackoff[T](minBackoff: FiniteDuration, maxBackoff: FiniteDuration, randomFactor: Double, onlyOnFailures: Boolean,
```

Comment:
This feature must be added in a backwards compatible way, i.e. a new method signature added and the old remaining, with deprecation annotation.

Reply:
I added a new method `onFailuresWithBackoff` that handles the scenario.
Thus, the `withBackoff` restarts on completes and failures, whereas the new method `onFailuresWithBackoff` restarts only on failures. Marking the original with deprecation annotation should be unnecessary now. However, the functionality of both could be merged into a single method next time backward compatibility could be broken.
Feel free to insist me on doing it using the deprecation way.

Reply:
ping @patriknw please whenever you are able, let me know what you think. Thanks.